### ISSUE TITLE
amplify env pull MaxListenersExceededWarning

### ISSUE THREAD

**Issue by zeroamplitude** [aws-amplify/amplify-cli#3304]
>**Note**: If your issue/bug is regarding the AWS Amplify Console service, please log it in the 
[Amplify Console GitHub Issue Tracker](https://github.com/aws-amplify/amplify-console/issues)

**Describe the bug**
When executing `amplify env pull` receiving warning `(node:60352) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 keypress listeners added. Use emitter.setMaxListeners() to increase limit`

**Amplify CLI Version**
4.13.1

**To Reproduce**
amplify env pull

**Expected behavior**
Fetch latest updates without errors.

**Screenshots**
![image](https://user-images.githubusercontent.com/2709598/73445789-755d7280-4329-11ea-9e0b-f9d72902b110.png)

**Desktop (please complete the following information):**
 - Windows
 - Node v8.10.0

**Comment by dantasfiles** [comment]
Getting the same thing with an `amplify init` with CLI v.4.13.1.
The problem does **not** occur with CLI v.4.13.0

- Run `npx react-native init EmitTest`
- Run `amplify init` with CLI v.4.13.1
```
> amplify init
Note: It is recommended to run this command from the root of your app directory
? Enter a name for the project EmitTest
? Enter a name for the environment env
? Choose your default editor: Visual Studio Code
? Choose the type of app that you're building javascript
Please tell us about your project
? What javascript framework are you using react-native
? Source Directory Path:  /
? Distribution Directory Path: /
? Build Command:  npm.cmd run-script build
? Start Command: npm.cmd run-script start
Using default provider  awscloudformation

For more information on AWS Profiles, see:
https://docs.aws.amazon.com/cli/latest/userguide/cli-multiple-profiles.html

? Do you want to use an AWS profile? Yes
? Please choose the profile you want to use (Use arrow keys)
> default (node:42892) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 keypress 
listeners added to [ReadStream]. Use emitter.setMaxListeners() to increase limit
```
Note that it says `11 keypress listeners added`, and it prompted me for user input 11 times before failing, so perhaps it has to do with whatever dependency Amplify is using to collect user input.

<details>
  <summary><strong>Environment</strong></summary>

```
  System:
    OS: Windows 10 10.0.18363
    CPU: (8) x64 Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz
    Memory: 1.92 GB / 7.85 GB
  Binaries:
    Node: 13.7.0 - C:\Program Files\nodejs\node.EXE
    npm: 6.13.6 - C:\Program Files\nodejs\npm.CMD
```
</details>

**Comment by UnleashedMind** [comment]
@dantasfiles 
We've seen this warning before that it was indeed due to the `inquier` package. 
Node.js 8.x is reaching end of life, have you tested it with newer versions of Node.js?

**Comment by dantasfiles** [comment]
@zeroamplitude is seeing the warning on Node 8. I'm seeing it with the following environment:
```
  System:
    OS: Windows 10 10.0.18363
    CPU: (8) x64 Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz
    Memory: 1.92 GB / 7.85 GB
  Binaries:
    Node: 13.7.0 - C:\Program Files\nodejs\node.EXE
    npm: 6.13.6 - C:\Program Files\nodejs\npm.CMD
```

**Comment by UnleashedMind** [comment]
@dantasfiles 
Thanks for the input

**Comment by zeroamplitude** [comment]
> @dantasfiles
> We've seen this warning before that it was indeed due to the `inquier` package.
> Node.js 8.x is reaching end of life, have you tested it with newer versions of Node.js?

@UnleashedMind I upgraded node to v12.14.1 and no further errors are showing.

Thanks

**Comment by zeroamplitude** [comment]
I spoke to soon.

It showed up again on `amplify env checkout master`

> node -v
v12.14.1

> amplify env checkout master
√ Initialized provider successfully.
? Do you want to configure Lambda Triggers for Cognito? (Y/n) (node:75040) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 end listeners added to [ReadStream]. Use emitter.setMaxListeners() t
o increase limit
(node:75040) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 keypress listeners added to [ReadStream]. Use emitter.setMaxListeners() to increase limit

**Comment by kaustavghosh06** [comment]
@zeroamplitude @dantasfiles Are you guys still seeing this issue?

**Comment by dantasfiles** [comment]
Let's run now:
```
> npx react-native init EventEmitterTest
...
> amplify init
Note: It is recommended to run this command from the root of your app directory
? Enter a name for the project EventEmitterTest
? Enter a name for the environment env        
? Choose your default editor: Visual Studio Code
? Choose the type of app that you're building javascript  
Please tell us about your project
? What javascript framework are you using react-native
? Source Directory Path:  /        
? Distribution Directory Path: /
? Build Command:  npm.cmd run-script build  
? Start Command: npm.cmd run-script start
Using default provider  awscloudformation

For more information on AWS Profiles, see:
https://docs.aws.amazon.com/cli/latest/userguide/cli-multiple-profiles.html

? Do you want to use an AWS profile? Yes
? Please choose the profile you want to use (Use arrow keys)
> default (node:26328) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 keypress listener? Please choose the profile you want to use default
\ Initializing project in the cloud...
```

Yep, still getting it.

Windows 10, Node v13.9.0

```
> npm -g ls inquirer
C:\Users\Daniel\AppData\Roaming\npm
`-- @aws-amplify/cli@4.13.4
  +-- amplify-category-analytics@2.13.3
  | `-- UNMET PEER DEPENDENCY inquirer@7.0.4  deduped
  +-- amplify-category-api@2.13.3
  | `-- UNMET PEER DEPENDENCY inquirer@7.0.4  deduped
  +-- amplify-category-auth@2.13.3
  | `-- UNMET PEER DEPENDENCY inquirer@7.0.4  deduped
  +-- amplify-category-function@2.13.3
  | `-- UNMET PEER DEPENDENCY inquirer@7.0.4  deduped
  +-- amplify-category-hosting@2.1.11
  | `-- UNMET PEER DEPENDENCY inquirer@7.0.4  deduped
  +-- amplify-category-interactions@2.1.11
  | `-- UNMET PEER DEPENDENCY inquirer@7.0.4  deduped
  +-- amplify-category-notifications@2.13.3
  | `-- UNMET PEER DEPENDENCY inquirer@7.0.4  deduped
  +-- amplify-category-predictions@2.1.12
  | `-- UNMET PEER DEPENDENCY inquirer@7.0.4  deduped
  +-- amplify-category-storage@2.1.11
  | `-- UNMET PEER DEPENDENCY inquirer@7.0.4  deduped
  +-- amplify-category-xr@2.1.11
  | `-- UNMET PEER DEPENDENCY inquirer@7.0.4  deduped
  +-- amplify-codegen@2.13.3
  | `-- UNMET PEER DEPENDENCY inquirer@7.0.4  deduped
  +-- amplify-frontend-android@2.13.2
  | `-- UNMET PEER DEPENDENCY inquirer@7.0.4  deduped
  +-- amplify-frontend-javascript@2.13.2
  | `-- UNMET PEER DEPENDENCY inquirer@7.0.4  deduped
  +-- amplify-provider-awscloudformation@4.13.4
  | `-- UNMET PEER DEPENDENCY inquirer@7.0.4  deduped
  `-- UNMET PEER DEPENDENCY inquirer@7.0.4

npm ERR! peer dep missing: inquirer@^5.0.0 || ^6.0.0, required by inquirer-autocomplete-prompt@1.0.2
npm ERR! peer dep missing: inquirer@^5.0.0 || ^6.0.0, required by inquirer-autocomplete-prompt@1.0.2
npm ERR! peer dep missing: inquirer@^5.0.0 || ^6.0.0, required by inquirer-autocomplete-prompt@1.0.2
npm ERR! peer dep missing: inquirer@^5.0.0 || ^6.0.0, required by inquirer-autocomplete-prompt@1.0.2
npm ERR! peer dep missing: inquirer@^5.0.0 || ^6.0.0, required by inquirer-autocomplete-prompt@1.0.2
npm ERR! peer dep missing: inquirer@^5.0.0 || ^6.0.0, required by inquirer-autocomplete-prompt@1.0.2
npm ERR! peer dep missing: inquirer@^5.0.0 || ^6.0.0, required by inquirer-autocomplete-prompt@1.0.2
npm ERR! peer dep missing: inquirer@^5.0.0 || ^6.0.0, required by inquirer-autocomplete-prompt@1.0.2
npm ERR! peer dep missing: inquirer@^5.0.0 || ^6.0.0, required by inquirer-autocomplete-prompt@1.0.2
npm ERR! peer dep missing: inquirer@^5.0.0 || ^6.0.0, required by inquirer-autocomplete-prompt@1.0.2
npm ERR! peer dep missing: inquirer@^5.0.0 || ^6.0.0, required by inquirer-autocomplete-prompt@1.0.2
npm ERR! peer dep missing: inquirer@^5.0.0 || ^6.0.0, required by inquirer-autocomplete-prompt@1.0.2
npm ERR! peer dep missing: inquirer@^5.0.0 || ^6.0.0, required by inquirer-autocomplete-prompt@1.0.2
npm ERR! peer dep missing: inquirer@^5.0.0 || ^6.0.0, required by inquirer-autocomplete-prompt@1.0.2
npm ERR! peer dep missing: inquirer@^5.0.0 || ^6.0.0, required by inquirer-autocomplete-prompt@1.0.2
```

**Comment by zeroamplitude** [comment]
> @zeroamplitude @dantasfiles Are you guys still seeing this issue?

Yes. The output is the same as @dantasfiles

**Comment by attilah** [comment]
@dantasfiles it is a dependency issue with Inquirer, in the meantime I'll create a fix which sets the max listener count to a high number. See my related comment [here](https://github.com/SBoudrias/Inquirer.js/issues/887#issuecomment-591736865)

**Comment by kesteinbakk** [comment]
I get same issue even after upgrading to latest version (4.13.4). How do I get the fix mentioned by @attilah ?

**Comment by attilah** [comment]
@kesteinbakk it will be included in the next release, 4.13.4 does not have it yet.
